File: fabric-mge/web/db/template.go
package db

import (
	"context"
	"log"
	"time"

	"go.mongodb.org/mongo-driver/bson"
)

type TemplateTable struct {
	Id          int       `bson:"_id"`
	Title       string    `bson:"title"`
	Category    string    `bson:"category"`
	Category_id int       `bson:"categoryId"`
	Abstract    string    `bson:"abstract_"`
	PubDate     time.Time `bson:"pubDate"`
	Published   bool      `bson:"published"`
	Content     string    `bson:"content"`
	ReviewState int       `bson:"reviewState"`
	Reviewer    string    `bson:"reviewer"`
	Author      string    `bson:"author"`
	Refcount    int       `bson:"refCount"`
	Method      int       `bson:"method"`
}

// type Content struct{

// }

// type TemplateTable struct {
// 	Id          int       `bson:"_id"`
// 	Title       string    `bson:"title"`
// 	Category    string    `bson:"category"`
// 	Category_id int       `bson:"categoryId"`
// 	Abstract    string    `bson:"abstract_"`
// 	PubDate     time.Time `bson:"pubDate"`
// 	Published   bool      `bson:"published"`
// 	Content     Content    `bson:"content"`
// 	ReviewState int       `bson:"reviewState"`
// 	Reviewer    string    `bson:"reviewer"`
// 	Author      string    `bson:"author"`
// 	Refcount    int       `bson:"refCount"`
// 	Method      int       `bson:"method"`
// }

func GetTemplateById(id int) TemplateTable {
	collection := db.Collection("template")
	var res TemplateTable
	err := collection.FindOne(context.Background(), bson.M{"_id": id}).Decode(&res)
	if err != nil {
		log.Fatal(err)
	}
	return res
}

func AddTemplate(t *TemplateTable) {
	collection := db.Collection("Template")
	objId, err := collection.InsertOne(context.TODO(), &t)
	if err != nil {
		log.Println(err)
		return
	}
	log.Println("录入数据成功，objId:", objId)
}

File: fabric-mge-backend/apps/storage/models/Template.py
from django.db import models
import mongoengine
from apps.account.models import Account, User
from apps.storage.models.Category import Category
import datetime


# Create your models here.


class Template(mongoengine.Document):
    id = mongoengine.SequenceField(primary_key=True)
    title = mongoengine.StringField(unique=True, max_length=255)
    # category = mongoengine.ReferenceField(Category)  # category_id
    category = mongoengine.StringField()  # 9 category_name
    # categoryId = mongoengine.ReferenceField(Category) # category_id
    categoryId = mongoengine.IntField(default=1)
    # abstract = mongoengine.StringField()
    abstract_ = mongoengine.StringField()
    pubDate = mongoengine.DateField(default=datetime.datetime.now())
    published = mongoengine.BooleanField(default=True)
    # content = mongoengine.DictField()
    content = mongoengine.StringField()
    reviewState = mongoengine.IntField(default=0)  # 0: 待审核   1: 审核通过 2: 审核不通过
    reviewer = ''
    author = mongoengine.ReferenceField(User)
    # author = mongoengine.StringField()
    refCount = mongoengine.IntField()
    method = mongoengine.IntField()
